ui/src/copyable_text.tsx

```diff
@@ -88,30 +92,34 @@ export const XCopyableText = ({ model }: { model: CopyableText }) => {
   React.useEffect(() => () => window.clearTimeout(timeoutRef.current), [])
 
   return (
-    <div
+    <Fluent.TextField
       data-test={name}
-      className={multiline ? css.multiContainer : css.compactContainer}
-      style={heightStyle as React.CSSProperties}
-    >
-      <Fluent.TextField
-        componentRef={ref}
-        value={value}
-        label={label}
-        multiline={multiline}
-        styles={{
-          root: { ...heightStyle, width: pc(100) },
-          wrapper: heightStyle,
-          fieldGroup: heightStyle || { minHeight: height },
-          field: { ...heightStyle, height, resize: multiline ? 'vertical' : 'none', },
-        }}
-        readOnly
-      />
-      <Fluent.PrimaryButton
-        title='Copy to clipboard'
-        onClick={onClick}
-        iconProps={{ iconName: copied ? 'CheckMark' : 'Copy' }}
-        className={clas(css.btn, copied ? css.copiedBtn : '', multiline ? css.btnMultiline : '')}
-      />
-    </div>
+      componentRef={ref}
+      value={value}
+      label={label}
+      multiline={multiline}
+      onRenderLabel={() =>
+        <div className={css.labelContainer}>
+          <Fluent.Label>{label}</Fluent.Label>
+          <Fluent.PrimaryButton
+            title='Copy to clipboard'
+            onClick={onClick}
+            iconProps={{ iconName: copied ? 'CheckMark' : 'Copy' }}
+            className={clas(css.btn, copied ? css.copiedBtn : '', multiline ? css.btnMultiline : '')}
+          />
+        </div>
+      }
+      styles={{
+        root: {
+          ...heightStyle,
+          ...styles.textFieldRoot,
+          ...multiline ? styles.textFieldMultiline : undefined
```

Comment:
Any particular reason for not inlining this?
```suggestion
          textFieldRoot: { position: 'relative', width: pc(100) },
          textFieldMultiline: multiline ? { '&:hover button': { opacity: 1 } } : undefined
```

Reply:
There is not. I try to prevent inlining styles whenever not necessary. But I agree this time the readability is better when inlined.